### PR TITLE
Restore backup UI

### PR DESCRIPTION
**Pull request by mikeger**
### Changes

The UI necessary to pick and initiate the restore of the history backup.

## Dependencies

Needs releases with:

- [ ] API changes to the SE

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.m [hunk 1]
```diff
@@ -45,6 +56,10 @@ - (void)viewDidLoad
     
     [self createViewConstraints];
     
+    if (self.contextType == ContextTypeNewDevice) {
+        [self createRestoreButton];
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.m]: I think the plan is to always show the button

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension NoHistoryViewController {
+    static let WireBackupUTI = "com.wire.ios-backup"
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift]: This should probably come from SE

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+extension NoHistoryViewController: UIDocumentMenuDelegate, UIDocumentPickerDelegate {
+    public func documentMenu(_ documentMenu: UIDocumentMenuViewController,
+                             didPickDocumentPicker documentPicker: UIDocumentPickerViewController) {
+        self.present(documentPicker, animated: true)
+    }
+    
+    public func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentAt url: URL) {
+
+        self.showLoadingView = true
+        // todo: read UUID()
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift]: The UUID check will be done in SE, the parameter will simply be removed

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.m (hunk 1 above)
Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.m]: Ah ok.

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift (hunk 2 above)
Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift]: You sure? We don't have it currently in the SE, but in Info.plist in the UI.

Comment (vytis) [reply Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift]: ah I understood this is backup extension, but this is more than that. Nevermind!

**Comment by mikeger** [comment]
@vytis sure, why not add some screenshots.

**Pull request merged**